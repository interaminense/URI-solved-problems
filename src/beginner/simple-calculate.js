/**
 * In this problem, the task is to read a code of a product 1, the number of units of product 1,
 * the price for one unit of product 1, the code of a product 2, the number of units of product 2
 * and the price for one unit of product 2. After this, calculate and show the amount to be paid.
 *
 * INPUT
 *
 * The input file contains two lines of data. In each line there will be 3 values: two integers and
 * a floating value with 2 digits after the decimal point.
 * @param {number} code
 * @param {number} count
 * @param {number} value
 *
 * OUTPUT
 *
 * The output file must be a message like the following example where "Value to pay". Remember the
 * space after ":" and after "R$" symbol. The value must be presented with 2
 * digits after the point.
 */

const tmpl = result => `R$ = ${result}`;

const calculate = (code, count, value) => count * value;

const valueToPay = (value1, value2) => tmpl((value1 + value2).toFixed(2));

export { calculate, tmpl, valueToPay };
