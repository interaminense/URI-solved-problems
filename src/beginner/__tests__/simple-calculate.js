import { calculate, tmpl, valueToPay } from '../simple-calculate';

describe('Simple calculate', () => {
  let value1 = 0;
  let value2 = 0;

  const result = (a, b) => valueToPay(a, b);

  it('The calculate must be 15.5', () => {
    value1 = calculate(12, 1, 5.3);
    value2 = calculate(16, 2, 5.1);

    expect(result(value1, value2)).toEqual(tmpl('15.50'));
    expect(result(value1, value2)).toMatchSnapshot();
  });

  it('The calculate must be 51.4', () => {
    value1 = calculate(13, 2, 15.3);
    value2 = calculate(161, 4, 5.2);

    expect(result(value1, value2)).toEqual(tmpl('51.40'));
    expect(result(value1, value2)).toMatchSnapshot();
  });

  it('The calculate must be 30.2', () => {
    value1 = calculate(1, 1, 15.1);
    value2 = calculate(2, 1, 15.1);

    expect(result(value1, value2)).toEqual(tmpl('30.20'));
    expect(result(value1, value2)).toMatchSnapshot();
  });
});
