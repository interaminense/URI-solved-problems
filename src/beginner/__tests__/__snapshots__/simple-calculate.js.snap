// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple calculate The calculate must be 15.5 1`] = `"R$ = 15.50"`;

exports[`Simple calculate The calculate must be 30.2 1`] = `"R$ = 30.20"`;

exports[`Simple calculate The calculate must be 51.4 1`] = `"R$ = 51.40"`;
